controllers/result.go

```diff
@@ -77,6 +80,52 @@ func (c *ResultController) Create() {
 	c.Redirect("/", http.StatusFound)
 }
 
+func (c *ResultController) Show() {
+	c.EnsureAuthenticatedUser()
+	c.TplName = "results/show.html"
+	c.Data["Title"] = "Result Detail"
+	web.ReadFromRequest(&c.Controller)
+
+	resultID := c.getResultID()
+	if resultID > 0 {
```

Comment:
1. Why do we need to check `resultID > 0` here? 
2. Also, this function does not have `else` while the function below has. I saw in the HTML file, I think cache page should also handle with the same behavior as show page (if given invalid result id should display some error to users instead of a blank page).

Reply:
> Why do we need to check resultID > 0 here?
Because in case that the given result id is not an integer the `c.getResultID()` won't be able to parse the id and will return the 0

> Also, this function does not have else while the function below has. I saw in the HTML file, I think cache page should also handle with the same behavior as show page (if given invalid result id should display some error to users instead of a blank page).
updated on [3842041](https://github.com/carryall/go-google-scraper-challenge/pull/72/commits/38420418c659cd7155899807afa3108c9f4bce2d)

Reply:
> Because in case that the given result id is not an integer the c.getResultID() won't be able to parse the id and will return the 0

It's kind of weird. How about let `getResultID()` function return both int64 and error instead of 0? I will be more clearer. What do you think?

Reply:
Ahh I see, updated in [5739193](https://github.com/carryall/go-google-scraper-challenge/pull/72/commits/5739193c3a6fecf60713b862f412034779b7c2a7), [e27ea2e](https://github.com/carryall/go-google-scraper-challenge/pull/72/commits/e27ea2e1bb2874cb880b8588f191043378074c25)